src/main/scala/beam/utils/scenario/UrbanSimScenarioLoader.scala

```diff
@@ -29,30 +31,40 @@ class UrbanSimScenarioLoader(
   val geo: GeoUtils
 ) extends LazyLogging {
 
+  implicit val ex: ExecutionContext = scala.concurrent.ExecutionContext.Implicits.global
+
   val population: Population = scenario.getPopulation
 
   val availableModes: String = BeamMode.allModes.map(_.value).mkString(",")
 
   def loadScenario(): Scenario = {
     clear()
 
-    val plans = scenarioSource.getPlans
-    logger.info(s"Read ${plans.size} plans")
-
-    val personsWithPlans = {
+    val plansF = Future {
+      val plans = scenarioSource.getPlans
+      logger.info(s"Read ${plans.size} plans")
+      plans
+    }
+    val personsF = Future {
       val persons: Iterable[PersonInfo] = scenarioSource.getPersons
       logger.info(s"Read ${persons.size} persons")
-      getPersonsWithPlan(persons, plans)
+      persons
+    }
+    val householdsF = Future {
```

Comment:
move this `Future` to `ScenarioSource`. Makes sense an adapter (`ScenarioSource`) returning a Future for expensive operation. In this case you have to change `BeamScenarioLoader` as well

Reply:
What is the purpose of moving this to `ScenarioSource`? Why should I break `ScenarioSource` interface? This was simple thing to do in `UrbanSimScenarioLoader`, otherwise good solution will be make it async all the way up to `BeamScenarioReader`

Reply:
The interface describes an Adapter to external source of information (in this case files xml/csv/parquet). It makes sense using futures to slow operations (like reading files, socket, database, etc.).
 At the same time we have redundancy: `BeamScenarioSource` and `UrbansimScenarioSource` that implements the same interface and could take advantage of Future approach.